pydantic_ai_slim/pydantic_ai/_utils.py

```diff
@@ -160,21 +160,23 @@ async def group_by_temporal(
             as soon as `aiter.__anext__()` returns. If `None`, no grouping/debouncing is performed
 
     Returns:
-        A context manager usable as an iterator async iterable of lists of items from the input async iterable.
+        A context manager usable as an async iterable of pairs of lists of items from the input async iterable,
+        and a boolean indicating whether the item was final coming out of the iterator.
     """
     if soft_max_interval is None:
 
-        async def async_iter_groups_noop() -> AsyncIterator[list[T]]:
+        async def async_iter_groups_noop() -> AsyncIterator[tuple[list[T], bool]]:
             async for item in aiter:
-                yield [item]
+                yield [item], False
+            yield [], True
```

Comment:
It's tempting to use `PeekableAsyncIterator` here to drop the final item, and we can do that, but with the way the code is written now, peeking updates the state of the streaming response, so in some scenarios if you call `.get()` etc. you get data from one step ahead of the events you've received. So I think we're better off just yielding a final extra item when there is no grouping.

Note that this is mostly a non-issue for normal usage where you would have grouping.